.github/workflows/windows.yml

```diff
@@ -74,3 +72,54 @@ jobs:
           Get-ChildItem env:
           cd $env:PYTHONPATH
           python -c "import sys; import scipy; sys.exit(not scipy.test())"
+
+
+  #############################################################################
+  full_dev_py_min_numpy:
+    name: cp39 (meson) full, dev.py, minimum numpy
+    if: "github.repository == 'scipy/scipy' || github.repository == ''"
+    runs-on: windows-2019
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          submodules: recursive
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.9'
+          cache: 'pip'
+          cache-dependency-path: 'environment.yml'
+
+      - name: Win_amd64 - install rtools
+        run: |
+          # mingw-w64
+          choco install rtools --no-progress
+          echo "c:\rtools40\ucrt64\bin;" >> $env:GITHUB_PATH
+
+      - name: Install OpenBlas
+        shell: bash
+        run: |
+          set -xe
+          bash tools/wheels/cibw_before_build_win.sh .
+          echo "PKG_CONFIG_PATH=c:\opt\64\lib\pkgconfig;" >> $GITHUB_ENV
+
+      - name: pip-packages
+        run: |
+          # 1.22.3 is currently the oldest numpy usable on cp3.9 according
+          # to pyproject.toml
+          python -m pip install numpy==1.22.3 cython pybind11 pythran meson-python meson ninja pytest pytest-xdist pytest-timeout pooch rich_click click doit pydevtool
+
+      - name: Build
+        run: |
+          $env:SCIPY_USE_PYTHRAN=1
+          python dev.py --build-dir=$pwd\build --install-prefix=$pwd\build-install build -j 2 --win-cp-openblas
```

Comment:
For some reason the defaults don't work here. When dev.py fires up meson I'm seeing a `meson needs absolute path for install-prefix` error. I'll make the change to remove these flags to illustrate.

Reply:
Must be a Windows-specific issue, I'll try and fix that separately.

Reply:
[4af90a6](https://github.com/scipy/scipy/pull/18325/commits/4af90a605c9ef63311b9dfc77d7125de5aa46bed) illustrates the issue. I amended [dev.py](https://github.com/scipy/scipy/pull/18325/files#diff-e42998d51257e6f84aa51ffa5f4ed1544cb12f97a94978c44f3bc281b5324d79R310) to print out the `Dirs.build` directory and the `Dirs.installed` directories. You can see from the [build log](https://github.com/scipy/scipy/actions/runs/4759018929/jobs/8457789963#step:7:19) that the `build` and `installed` directories aren't being resolved to absolute paths. I think this is because the build directory hasn't been created when the [resolve](https://github.com/scipy/scipy/blob/main/dev.py#L309) occurs. Meson doesn't like relative paths. Perhaps there are differences between cp39/cp3X and Win/Linux in the path resolution?

Error message:

```
meson.build:1:0: ERROR: prefix value 'build-install' must be an absolute path

A full log can be found at D:\a\scipy\scipy\build\meson-logs\meson-log.txt
Meson build setup failed!
Traceback (most recent call last):
  File "D:\a\scipy\scipy\tools\openblas_support.py", line [37](https://github.com/scipy/scipy/actions/runs/4759018929/jobs/8457789963#step:7:38)2, in <module>
    make_init(args.write_init[0])
  File "D:\a\scipy\scipy\tools\openblas_support.py", line 232, in make_init
    with open(os.path.join(dirname, '_distributor_init.py'), 'w') as fid:
FileNotFoundError: [Errno 2] No such file or directory: 'D:\\a\\scipy\\scipy\\build-install\\Lib\\site-packages\\scipy\\_distributor_init.py'
```

Reply:
The solution will be to amend `Dirs.__init__` in dev.py to definitively create absolute paths. It doesn't happen in the other Windows meson job, so it's localised to cp39.

Reply:
Ah yes, looks like it's a `pathlib` bug. It had a bunch of issues with `.resolve()`, `.absolute()`, etc. that were only recently resolved.

Reply:
Yes, that sounds fine to me.